Modules/Sources/WordPressUI/Views/AdaptiveTabBar.swift

```diff
@@ -0,0 +1,240 @@
+import UIKit
+import WordPressShared
+
+public protocol AdaptiveTabBarItem {
+    var localizedTitle: String { get }
+}
+
+public class AdaptiveTabBar: UIControl {
+
+    // MARK: - UI Components
+
+    private lazy var scrollView: UIScrollView = {
+        let scrollView = UIScrollView()
+        scrollView.translatesAutoresizingMaskIntoConstraints = false
+        scrollView.showsHorizontalScrollIndicator = false
+        scrollView.showsVerticalScrollIndicator = false
+        scrollView.scrollsToTop = false
+        return scrollView
+    }()
+
+    private lazy var stackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .horizontal
+        stackView.distribution = .fillEqually
+        stackView.spacing = 0
+        return stackView
+    }()
+
+    private lazy var selectionIndicator: UIView = {
+        let view = UIView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.backgroundColor = .label
+        view.layer.cornerRadius = 2
+        view.isHidden = true
+        return view
+    }()
+
+    // MARK: - Properties
+
+    var items: [AdaptiveTabBarItem] = [] {
+        didSet { refreshTabs() }
+    }
+
+    private var buttons: [UIButton] = []
+
+    private(set) var selectedIndex: Int = 0 {
+        didSet {
+            buttons[oldValue].isSelected = false
+            buttons[selectedIndex].isSelected = true
+        }
+    }
+
+    private var widthConstraint: NSLayoutConstraint!
+    private var indicatorWidthConstraint: NSLayoutConstraint?
+    private var indicatorCenterXConstraint: NSLayoutConstraint?
+
+    public let tabBarHeight: CGFloat = 44
+
+    // MARK: - Initialization
+
+    public init() {
+        super.init(frame: .zero)
+        setup()
+    }
+
+    public required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        setup()
+    }
+
+    private func setup() {
+        addSubview(scrollView)
+        scrollView.pinEdges()
+
+        scrollView.addSubview(stackView)
+        stackView.pinEdges()
+
+        NSLayoutConstraint.activate([
+            heightAnchor.constraint(equalToConstant: tabBarHeight),
+            stackView.heightAnchor.constraint(equalTo: heightAnchor)
+        ])
+
+        widthConstraint = stackView.widthAnchor.constraint(equalTo: scrollView.widthAnchor)
```

Comment:
I believe `stackView.pinEdges()` also adds horizontal constraints that affect `stackView.width` and `scrollView.width`. Would this `widthConstraint` cause conflicts with the constraints added by the `pinEdges`?

Reply:
I think it works as designed. It ensures that the content size matches the scroll size when there is enough space (so that it takes full width). When the content doesn't match, it becomes scrollable:


<img width="456" height="972" alt="Screenshot 2025-09-03 at 12 18 45 PM" src="https://github.com/user-attachments/assets/b6a398d9-3228-4de8-b208-d51e876ae8f2" />